### ISSUE TITLE
Access underlying parser settings

### ISSUE THREAD

**Issue by XapaJIaMnu** [databricks/spark-csv#353]
Hey,

I have a CSV which contains fields with very long lines. and I get the following exception:
`
Exception in task 0.0 in stage 1.0 (TID 1)
com.univocity.parsers.common.TextParsingException: Length of parsed input (100001) exceeds the maximum number of characters defined in your parser settings (100000). 
Hint: Number of characters processed may have exceeded limit of 100000 characters per column. Use settings.setMaxCharsPerColumn(int) to define the maximum number of characters a column can have
Ensure your configuration is correct, with delimiters, quotes and escape sequences that match the input format you are trying to parse
`
The setting in question is found at: [https://github.com/databricks/spark-csv/blob/master/src/main/scala/com/databricks/spark/csv/readers/readers.scala#L62](url). As far as I can tell there isn't a way to set it since the parser is not exposed (correct me if I am wrong, I am a beginner in scala).  Is there a way to setting it other than editing the spark-csv code?

Cheers,

Nick

**Comment by HyukjinKwon** [comment]
I guess this is https://issues.apache.org/jira/browse/SPARK-14103 maybe.

**Comment by HyukjinKwon** [comment]
Oh, sorry I missed " very long lines.". Yes it is not configurable here.

**Comment by XapaJIaMnu** [comment]
That should be possible in the future? It can't be expected that the defaults would work for everyone?

**Comment by HyukjinKwon** [comment]
I think it should be configurable too. I am not sure (I can't decide) for this library because I guess this library would not be updated actively because this one is absorbed into Spark (from 2.0).

So, I actually submitted a JIRA before to increase this value here, https://issues.apache.org/jira/browse/SPARK-14260 but it was kind of rejected. 

I think you can submit another issue for Spark because it seems clearly an issue although the priority can be low.

**Comment by HyukjinKwon** [comment]
How about changing your parse library to Apache common one? I guess that does not have such a limit.

**Comment by XapaJIaMnu** [comment]
Yeah, I think I would do this for now, it's just that AFAIK univocity is faster and I am dealing with hundreds of gigabytes worth of CSVs, so I was hoping to use the faster one.

**Comment by jbax** [comment]
Just a heads up: from version 2.2.0 you can set the limit to -1 which configures the parser to auto-expand the internal buffer. You can also set the error content length to limit or omit the parsed content being displayed in error messages.

**Comment by HyukjinKwon** [comment]
@jbax Cool!

**Comment by dhruvgairola** [comment]
@jbax How can you set this limit for univocity in spark-csv? I don't see any such option in their Github documentation. Thanks in advance!

**Comment by jbax** [comment]
@dhruvgairola , yes `option("maxCharsPerColumn", "-1")` should do the trick.

**Comment by SindhujaM** [comment]
Hi..

I am having some data in a CSV format (approx. 1.5GB)

When I load this in spark I am getting the below mentioned error. But this works fine for data in MB.

My sample file Loaded to dataFrame 

df = sqlContext.read().format("com.databricks.spark.csv")
					.option("header", "true")
					.option("delimiter", ",")
					.option("parserLib", "univocity")
					.load("/home/mycsvFile.csv").toDF();
df.write().format("com.databricks.spark.csv").option("header", "true")
		.save("Myoutputfile.csv"));
		System.out.println("File is saved sucessfully");

ERROR: org.apache.spark.executor.Executor - Exception in task 1.0 in stage 1.0 (TID 2)
com.univocity.parsers.common.TextParsingException: Length of parsed input (100001) exceeds the maximum number of characters defined in your parser settings (100000). 
Identified line separator characters in the parsed content. This may be the cause of the error. The line separator in your parser settings is set to '\n'. Parsed content: [url=http://forums.hardwarezone.com.sg/member.php?u=422183]HaoGuShiXiongDi[/url] has reported a post.[\n]................................
..............................
...............................................
......................


Kindly help me to resolve this issue..

Thanks in advance..!!

**Comment by SindhujaM** [comment]
How to resolve ???

ERROR: org.apache.spark.executor.Executor - Exception in task 0.0 in stage 1.0 (TID 1)
com.univocity.parsers.common.TextParsingException: Length of parsed input (100001) exceeds the maximum number of characters defined in your parser settings (100000). 
Identified line separator characters in the parsed content. This may be the cause of the error. The line separator in your parser settings is set to '\n'. Parsed content: [url=http://forums.hardwarezone.com.sg/member.php?u=422183]HaoGuShiXiongDi[/url] has reported a post.[\n]
	[\n]
	Reason:[quote]Spamming a wall of text.[\n]....
......................
..........................

I have loaded 1.5 GB of my CSV file in DataFrame . But getting Lenght exceed exception. Kindly help me to resolve this..

MY Sample

DataFrame df = sqlContext.read().format("com.databricks.spark.csv")
				.option("header", "true")
				.option("inferSchema", "true")
				.option("delimiter", ",")
				.option("escape", ";")
				.option("parserLib", "UNIVOCITY")
				.load(input).toDF();

df.write().format("com.databricks.spark.csv").option("header", "false")
		.save("output.csv");

**Comment by falaki** [comment]
This option is exposed in the inline csv parser that ships with Spark 2.x. I recommend you switch to Spark 2.x.
